v2/protocol/http/utility.go

```diff
@@ -24,3 +26,63 @@ func NewEventFromHTTPResponse(resp *nethttp.Response) (*event.Event, error) {
 	msg := NewMessageFromHttpResponse(resp)
 	return binding.ToEvent(context.Background(), msg)
 }
+
+// NewEventsFromHTTPRequest returns a batched set of Events from a http.Request
+func NewEventsFromHTTPRequest(req *nethttp.Request) ([]event.Event, error) {
+	msg := NewMessageFromHttpRequest(req)
+	return binding.ToEvents(context.Background(), msg, msg.BodyReader)
```

Comment:
Question: use req.Context() here? Not sure if that provides value/advantage in this case.

Reply:
I was being consistent with all the other utility functions as previously written (they all used `context.Background()`. My suggestion would e that if we want to make that change it should probably be in a separate PR.